### PR TITLE
Code reorganization

### PR DESCRIPTION
**Pull request by Khyme**

### CODE REVIEW

File: internal/provider/datasource/service/service_data_source.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
```
[the remark this answers is not in the document]
Comment (Khyme) [reply internal/provider/datasource/service/service_data_source.go:10]: that sounds reasonable ;)

File: internal/provider/provider.go [hunk 2]
```diff
@@ -14,6 +14,10 @@ import (
 	"github.com/hashicorp/terraform-plugin-log/tflog"
 
 	tsClient "github.com/timescale/terraform-provider-timescale/internal/client"
+	d_products "github.com/timescale/terraform-provider-timescale/internal/provider/datasource/products"
```
Comment (alevy713) [inline internal/provider/provider.go:17]: I'm a little conflicted on having the duplicated packages with different import paths. WDYT of this setup

```
provider/
    products/
        data_source.go
    service/
        resource.go
        datasource.go
    vpcs/
         resource.go
         datasource.go
   provider.go
```

File: internal/provider/provider.go [hunk 3]
```diff
@@ -120,17 +124,17 @@ func (p *TimescaleProvider) Configure(ctx context.Context, req provider.Configur
...
 }
 
 // DataSources defines the data sources implemented in the provider.
 func (p *TimescaleProvider) DataSources(ctx context.Context) []func() datasource.DataSource {
 	tflog.Trace(ctx, "TimescaleProvider.DataSources")
 	return []func() datasource.DataSource{
-		NewProductsDataSource,
-		NewServiceDataSource,
-		NewVpcsDataSource,
+		d_products.NewProductsDataSource,
```
Comment (alevy713) [inline internal/provider/provider.go:135]: now that everything is in packages we can remove the Timescale naming from everything. so if we stay with this it'd be `d_products.NewDataSource` or with the other suggestion `products.NewDataSource`

**Review by alevy713 — COMMENTED** [review]
I like the idea of grouping related items together into packages. I left some suggestions that we could group by Timescale identities instead of Hashicorp ones.

**Pull request closed without merging**